Apps/GatewayApi/src/Services/UserEmailService.cs

```diff
@@ -266,8 +254,8 @@ private async Task AddVerificationEmailAsync(string hdid, string toEmail, Guid i
                 [EmailTemplateVariable.ExpiryHours] = verificationExpiryHours.ToString("0", CultureInfo.CurrentCulture),
             };
 
-            EmailTemplate emailTemplate = await this.emailQueueService.GetEmailTemplateAsync(EmailTemplateName.RegistrationTemplate, ct) ?? throw new ProblemDetailsException(
-                ExceptionUtility.CreateServerError($"{ServiceType.Database}:{ErrorType.CommunicationInternal}", ErrorMessages.EmailTemplateNotFound));
+            EmailTemplate emailTemplate = await this.emailQueueService.GetEmailTemplateAsync(EmailTemplateName.RegistrationTemplate, ct) ??
+                                          throw new NotFoundException(ErrorMessages.EmailTemplateNotFound);
```

Comment:
Affects:

- GatewayApi
  - UserProfileController.CloseUserProfile()
  - UserProfileController.RecoverUserProfile()

This is the same method that deals with queuing emails where I suggested a DatabaseException instead of NotFoundException.

Reply:
This exception should propagate to the surface and should be a 500 (potentially a database error)

Reply:
Similar to the other getEmailTemplateAsync call I've decided internally this is a database related exception in the sense that the data for the process is missing and not found means nothing to the process in question.